swim.go

```diff
@@ -45,21 +52,51 @@ type SWIM struct {
 	// Currently connected memberList
 	memberMap *MemberMap
 
+	messageEndpoint *MessageEndpoint
+
+	priorityPBStore *PriorityPBStore
+
 	// FailureDetector quit channel
 	quitFD chan struct{}
 }
 
-func New(config *Config) *SWIM {
-
+func New(config *Config, messageEndpointConfig MessageEndpointConfig, awareness *Awareness) *SWIM {
 	if config.T < config.AckTimeOut {
 		panic("T time must be longer than ack time-out")
 	}
 
-	return &SWIM{
-		config:    config,
-		memberMap: NewMemberMap(),
-		quitFD:    make(chan struct{}),
+	swim := SWIM{
+		config:          config,
+		memberMap:       NewMemberMap(),
+		messageEndpoint: nil,
+		priorityPBStore: NewPriorityPBStore(config.MaxlocalCount),
+		quitFD:          make(chan struct{}),
 	}
+
+	messageEndpoint := setMessageEndpoint(config, messageEndpointConfig, &swim, awareness)
+	swim.messageEndpoint = messageEndpoint
+
+	return &swim
+}
+
+// set MessageEndpoint
+func setMessageEndpoint(config *Config, messageEndpointConfig MessageEndpointConfig, messageHandler MessageHandler, awareness *Awareness) *MessageEndpoint {
+	packetTransportConfig := PacketTransportConfig{
```

Comment:
why don't you just integrate this function into `New` or I think this function name would be better like `messageEndpointFactory`. I think this function not set MessageEndpoint to swim just create and return

Reply:
you mean, you think that it is better to implement into `New` function at once?

Reply:
oh.. swim is used for creating `MessageEndpoint`.. then I think it is better to change func name